libs/sdk-backend-tiger/src/convertors/toBackend/afm/TotalsConverter.ts

```diff
@@ -41,6 +51,125 @@ export function convertTotals(def: IExecutionDefinition): Total[] {
             return tigerTotal;
         });
     }).filter((total): total is Total => total !== undefined);
+
+    /**
+     * With new column totals, we need to have in consideration the cases where same totals/subtotals are selected for rows and columns.
+     * This new cases would require extra configuration so Tiger AFM is able to compute properly the result.
+     * These new cases are -
+     *   1. Grand totals - is the value obtained from row and column totals.
+     *   2. Marginal totals - is the value obtained within the subgroups from row and column subtotals.
+     *
+     * For `Grand Totals`, in Tiger AFM, you specify that you want total of totals with total that have only "measureGroup" present.
+     * For `Marginal Total`, in Tiger AFM, would need to iterate through both dimensions and obtain the missing totalDimensions items
+     * based on the attribute and column identifiers order in buckets.
+     */
+
+    const attributeBucket = bucketsFind(buckets, "attribute");
+    const columnBucket = bucketsFind(buckets, "columns");
+
+    const matchingAttributeBucket = attributeBucket ? bucketAttributes(attributeBucket) : [];
+    const matchingColumnBucket = columnBucket ? bucketAttributes(columnBucket) : [];
+
+    let newTotals = totals;
+
+    if (shouldAddGrandTotalOrMarginalTotal(attributeBucket, columnBucket)) {
+        const attributeIdentifiers = getAttributeBucketIdentifiers(matchingAttributeBucket);
+        const columnIdentifiers = getAttributeBucketIdentifiers(matchingColumnBucket);
+
+        attributeBucket!.totals?.forEach((attributeTotal, index) => {
+            let hasRowAndColumnGrandTotals = false;
+            let hasRowAndColumnSubTotals = false;
+            let attributeDimensionIndex = 0;
+            let columnDimensionIndex = 0;
+            columnBucket!.totals?.forEach((columnTotal, columnIndex) => {
+                const attributeMeasureId = attributeTotal.measureIdentifier;
+                const attributeType = attributeTotal.type;
+                const columnMeasureId = columnTotal.measureIdentifier;
+                const columnType = columnTotal.type;
+                // Check for totals from same measure and type
+                if (
+                    attributeAndColumnHasSameTotal(
+                        attributeMeasureId,
+                        attributeType,
+                        columnMeasureId,
+                        columnType,
+                    )
+                ) {
+                    // Check for grand totals rows/columns
```

Comment:
consider appending this to the comment to be more clear: `Grand total is always defined on the very first attribute of the attribute/columns bucket`

Reply:
Added.